src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Android.cs

```diff
@@ -0,0 +1,409 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace System
+{
+    public sealed partial class TimeZoneInfo
+    {
+        private const string TimeZoneFileName = "tzdata";
+
+        private static AndroidTzData tzData = new AndroidTzData();
```

Comment:
Could it be allocated lazily? On mobile devices it's better to do as little as possible, and the constructor for AndroidTzData interates over several paths and loads potentially a lot of data. Applications shouldn't need to pay the price for it until it's absolutely needed.

Reply:
Thanks, how does it look now?